subsys/bluetooth/mesh/blob_srv.c

```diff
@@ -844,6 +844,19 @@ static int blob_srv_settings_set(struct bt_mesh_model *mod, const char *name,
 		return 0;
 	}
 
+	/* If device restarted before it handled `XFER_START` server doesn't have transfer
+	 * to be restored - transfer should be abandoned and new one must be started
+	 */
+	if (srv->state.xfer.size == 0) {
+		LOG_DBG("Transfer from 0x%04x (%llu) hasn't started, abandoning", srv->state.cli,
+			srv->state.xfer.id);
+		phase_set(srv, BT_MESH_BLOB_XFER_PHASE_INACTIVE);
+		srv->state.xfer.mode = BT_MESH_BLOB_XFER_MODE_NONE;
+		srv->state.ttl = BT_MESH_TTL_DEFAULT;
+		erase_state(srv);
```

Comment:
Not sure if it is correct to store something in settings subsystem while it loads the data. I'd suggest to either leave the stored value or handle this in `blob_srv_start`: https://github.com/zephyrproject-rtos/zephyr/blob/9ca9b91b1330d982ef24f610992682951ca85934/subsys/bluetooth/mesh/blob_srv.c#L885-L891